core/src/main/scala/cats/ContravariantMonoidal.scala

```diff
@@ -11,12 +11,24 @@ import simulacrum.typeclass
  * Based on ekmett's contravariant library:
  * https://hackage.haskell.org/package/contravariant-1.4/docs/Data-Functor-Contravariant-Divisible.html
  */
-@typeclass trait ContravariantMonoidal[F[_]] extends ContravariantSemigroupal[F] { self =>
+@typeclass trait ContravariantMonoidal[F[_]] extends ContravariantSemigroupal[F] with InvariantMonoidal[F] {
   /**
    * `unit` produces an instance of `F` for any type `A`
    * that is trivial with respect to `contramap2` along
    * the diagonal
    */
   def unit[A]: F[A]
+
+  override def pure[A](a: A): F[A] = unit
+
+  override def monoid[A](implicit A: Monoid[A]): Monoid[F[A]] =
+    ContravariantMonoidal.monoid(this)
```

Comment:
this seems to be breaking the previous pattern: we are calling the object from the trait, but in the other cases we call it the opposite direction.

Reply:
Yes, this is because we can define a `ContravariantMonoidal` regardless of if `A` is a Monoid or not.